### PR TITLE
Correct update section for Trento Server and Agents

### LINKED ISSUE #107: Trento premium: update sections outdated
The update sections in the Trento Premium documentation still reflect the update process for the community / open-source version. They must be updated and aligned with the Trento Premium installation process.

Particularly:

- In the case of the Trento Server, the update process should consist of two steps:
Step 1: `export KUBECONFIG=/etc/rancher/k3s/k3s.yaml`
Step 2: `HELM_EXPERIMENTAL_OCI=1 helm upgrade --install trento-server oci://registry.suse.com/trento/trento-server --version 0.3.5 --set-file trento-runner.privateKey=PRIVATE_SSH_KEY`
- In the case of the Trento Agent, the update process is identical to the installation process.

### PR DESCRIPTION
**Pull request by tomschr**
This PR fixes #107 

**Hint**: Although correct, the diff might be a bit confusing. I had to rework the procedures for the Updating the Trento Agents a bit as copy-and-paste does not really work.

It might be easier to review the screenshots below.

<details>
  <summary>Section Updating Trento Server (click me)</summary>

Screenshot 2022-02-03 at 10-49-41 Getting started with Trento Premium
</details>

<details>
  <summary>Section Updating Trento Agents (click me)</summary>

Screenshot 2022-02-03 at 10-49-57 Getting started with Trento Premium
</details>

### CODE REVIEW

File: xml/article_sap_trento.xml [hunk 1]
```diff
@@ -928,13 +929,26 @@ Warning: Permanently added '<replaceable>TRENTO_AGENT_IP</replaceable>' (ECDSA)
         <para> Ensure the &t.server; is running. </para>
       </step>
       <step>
-        <para> On the &t.server; host start the update: </para>
-        <screen>&trentoagentprompt;sudo curl -sfL https://raw.githubusercontent.com/trento-project/trento/main/install-server.sh | bash
+        <para>Export the <envar>KUBECONFIG</envar> environment variable for the
+          same user that installed K3s:
+        </para>
+        <screen>export KUBECONFIG=/etc/rancher/k3s/k3s.yaml</screen>
```
Comment (ajaeger) [inline xml/article_sap_trento.xml:935]: This is k3s specific.

I would reword it as:
Ensure that credentials for Kubernetes are set up correctly. For example, if you run K3s, export the KUBECONFIG ...

File: xml/article_sap_trento.xml [hunk 2]
```diff
@@ -928,13 +929,26 @@ Warning: Permanently added '<replaceable>TRENTO_AGENT_IP</replaceable>' (ECDSA)
...
+      </step>
+      <step>
+        <para>
+          With the same user that installed K3s, start the update
```
Comment (ajaeger) [inline xml/article_sap_trento.xml:939]: This needs to  work with every kubernetes, suggestion:

With the same user that installed Trento server initially, start the update of the Trento server.

Btw. remove host - that's k3s specific.

File: xml/article_sap_trento.xml [hunk 3]
```diff
@@ -928,13 +929,26 @@ Warning: Permanently added '<replaceable>TRENTO_AGENT_IP</replaceable>' (ECDSA)
...
+   oci://registry.suse.com/trento/trento-server \
+   --version 0.3.5 \
+   --set-file trento-runner.privateKey=<replaceable>PRIVATE_SSH_KEY</replaceable>
         </screen>
       </step>
       <step>
         <para>Check the &t.server; processes:</para>
-        <screen>&trentoagentprompt;sudo kubectl get pod</screen>
+        <screen>&trentoagent.prompt;sudo kubectl get pod</screen>
       </step>
```
Comment (ajaeger) [inline xml/article_sap_trento.xml:952]: Why is Root needed here? That should not be needed, the helm upgrade and the kubectl are run as same user without root needed.

Remove sudo

File: xml/article_sap_trento.xml [hunk 4]
```diff
@@ -962,45 +985,49 @@ Warning: Permanently added '<replaceable>TRENTO_AGENT_IP</replaceable>' (ECDSA)
...
+&trentoagent.prompt;sudo zypper install trento-premium</screen>
       </step>
+      <!--<step>
+        <para>
+          Open the configuration file <filename>/etc/trento/agent.yml</filename>
+          and uncomment (<literal>#</literal>) the entries for
+          <parameter>ssh-address</parameter> and <parameter>collector-host</parameter>
+          and update the values appropriately.
+        </para>-->
+      <!--</step>-->
```
Comment (ajaeger) [inline xml/article_sap_trento.xml:1001]: Why is this added here? I suggest to remove.

File: xml/article_sap_trento.xml [hunk 5]
```diff
@@ -962,45 +985,49 @@ Warning: Permanently added '<replaceable>TRENTO_AGENT_IP</replaceable>' (ECDSA)
...
       <step>
-        <para>Update &t.agent;, using the &t.agent; host local IP and
-          the &t.server; host local IP:</para>
-        <screen>&trentoagent.prompt;curl -sfL https://raw.githubusercontent.com/trento-project/trento/main/install-agent.sh | sudo bash -s - --agent-bind-ip=<replaceable>TRENTO_AGENT_IP</replaceable> --server-ip=<replaceable>TRENTO_SERVER_IP</replaceable></screen>
+        <para>
+          Start the &t.agent;:
+        </para>
+        <screen>&trentoagent.prompt;sudo systemctl enable --now trento-agent</screen>
       </step>
```
Comment (ajaeger) [inline xml/article_sap_trento.xml:1007]: AFAIK this should be:

systemctl restart trento-agent

**Review by ajaeger — CHANGES_REQUESTED** [review]

File: xml/article_sap_trento.xml [hunk 6]
```diff
@@ -962,45 +985,49 @@ Warning: Permanently added '<replaceable>TRENTO_AGENT_IP</replaceable>' (ECDSA)
...
+          <parameter>ssh-address</parameter> and <parameter>collector-host</parameter>
+          and update the values appropriately.
+        </para>-->
+      <!--</step>-->
       <step>
-        <para>Update &t.agent;, using the &t.agent; host local IP and
-          the &t.server; host local IP:</para>
-        <screen>&trentoagent.prompt;curl -sfL https://raw.githubusercontent.com/trento-project/trento/main/install-agent.sh | sudo bash -s - --agent-bind-ip=<replaceable>TRENTO_AGENT_IP</replaceable> --server-ip=<replaceable>TRENTO_SERVER_IP</replaceable></screen>
+        <para>
+          Start the &t.agent;:
```
Comment (ajaeger) [inline xml/article_sap_trento.xml:1004]: Start -> Restart

**Review by abravosuse — COMMENTED** [review]
@tomschr @ajaeger Sorry but I'm unable to follow the sequence of changes. Would it be possible to see a preview to understand what it looks like with the current changes?

File: xml/article_sap_trento.xml (hunk 3 above)
Comment (tomschr) [reply xml/article_sap_trento.xml:952]: Remember, root != trentoadmin. :slightly_smiling_face: I can remove the prompt and the sudo.

File: xml/article_sap_trento.xml (hunk 4 above)
Comment (tomschr) [reply xml/article_sap_trento.xml:1001]: Removed.

File: xml/article_sap_trento.xml (hunk 5 above)
Comment (tomschr) [reply xml/article_sap_trento.xml:1007]: Ok, done

File: xml/article_sap_trento.xml (hunk 6 above)
Comment (tomschr) [reply xml/article_sap_trento.xml:1004]: Done

File: xml/article_sap_trento.xml (hunk 2 above)
Comment (tomschr) [reply xml/article_sap_trento.xml:939]: Ok, done.

File: xml/article_sap_trento.xml (hunk 1 above)
Comment (tomschr) [reply xml/article_sap_trento.xml:935]: done

**Comment by tomschr** [comment]
@abravosuse No problem, Alberto. I've added the screenshot of the two sections:

<details>
  <summary>Updating Trento Server</summary>

Screenshot 2022-02-04 at 09-26-25 Getting started with Trento Premium
</details>

<details>
  <summary>Updating Trento Agents</summary>

Screenshot 2022-02-04 at 09-26-48 Getting started with Trento Premium

</details>

**Review by ajaeger — COMMENTED** [review]
Thanks, this looks fine to me!

**Review by ajaeger — COMMENTED** [review]
Btw. double checking I found one more problem, both update and install have:

"Loaded: loaded (/usr/lib/systemd/system/trento-agent.service; disabled; vendor preset: disabled)"

It should be "Loaded: loaded (/usr/lib/systemd/system/trento-agent.service; enabled; vendor preset: disabled)"

since step 5 of install runs " systemctl enable --now trento-agent"

**Comment by tomschr** [comment]
Thanks Andreas, I've added them into the doc.

**Review by ajaeger — COMMENTED** [review]
Thanks!
